pkg/sidecar_mounter/sidecar_mounter.go

```diff
@@ -213,25 +214,34 @@ func logVolumeTotalSize(dirPath string) {
 }
 
 // collectMetrics collects metrics from the gcsfuse instance every 10 seconds.
-func collectMetrics(ctx context.Context, port, dirPath string) {
+func collectMetrics(ctx context.Context, port, dirPath string, scrapeInterval int) {
+	genNumber := 1
 	metricEndpoint := "http://localhost:" + port + "/metrics"
-	outputPath := dirPath + "/metrics.prom"
-	ticker := time.NewTicker(10 * time.Second)
+	ticker := time.NewTicker(time.Duration(scrapeInterval) * time.Second)
 
 	for {
 		select {
 		case <-ctx.Done():
 			return
 		case <-ticker.C:
 			newCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
-			scrapeMetrics(newCtx, metricEndpoint, outputPath)
+			scrapeMetrics(newCtx, metricEndpoint, dirPath, genNumber)
+			// If x is the current generation of the metrics snapshot/file, we are leaving some buffer where the outdated (x-1)
+			// file exists so that the reader has time to finish reading the complete data.
+			// The -2 means we are deleting file generations up to x - 2.
+			deleteOutdatedFile(dirPath, genNumber-2)
```

Comment:
This only deletes one outdated file "genNumber-2", but if this deletion keep failing, we will have prom file leak on the disk.

Can we try to delete all the outdated files each time? For example, delete all the files with genNumber <= latest genNumber-2

Reply:
Since we are listing the directory contents already, we can delete all the outdated metrics files that we find there.